compiler/shared_ast/typing.mli

```diff
@@ -61,9 +61,12 @@ val expr :
     filling the gaps ([TAny]) if any. Use [Expr.untype] first if this is not
     what you want.
 
-    Note that typing also transparently performs disambiguation of constructors:
-    [EDStructAccess] nodes are translated into [EStructAccess] with the suitable
-    structure and field idents (this only concerns [desugared] expressions). *)
+    Note that typing also transparently performs
+    - disambiguation of constructors: [EDStructAccess] nodes are translated into
+      [EStructAccess] with the suitable structure and field idents (this only
+      concerns [desugared] expressions).
+    - resolution of operator types, which are stored (monomorphised) back in the
```

Comment:
They are stored in `EAbs`, `EApp` and `EAppOp` then?

Reply:
Indeed `EApp` should be added to this list, I'll clarify this comment. `EAbs` nodes can be changed as well when their types wasn't fully specified (it may happen from generated code only), but that was already the case.